src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage;
```

Comment:
It seems that the `Storage` instance should not be passed into the `ShopServiceImpl` as a dependency. Instead, the `Storage` should be accessed directly in the operation handlers. This approach adheres to the SOLID principles, particularly the single responsibility principle, by making `ShopServiceImpl` responsible only for orchestrating the strategy execution and not for maintaining the state of the storage.